### PR TITLE
Fix/applics 509 remediation

### PR DESCRIPTION
**Pull request by HadrianHughesKC**
## Describe your changes

* feat(web/applications): disable purple visited links across the app
* fix(web/applications): ensure 'Unpublish project' link is always red
* fix(web/applications): change button text for unpublished case changes
* fix(api/applications): disallow publishing S51 without Welsh fields
* refactor(api/applications): return instead of throw from validator
* fix(web/applications): use correct error when S51 fields are missing
* fix(web/applications): align case title edit page with latest designs
* refactor(web/applications): use SummaryList component on Overview page
* fix(web/applications): use textarea for Webfilter (English & Welsh)
* test(web/applications): update tests and snapshots
* feat(web/applications): update project name title to match designs
* test(e2e): change title/label for the project name field
test(e2e): expect a textarea for the Webfilter field
refactor(e2e): encapsulate computation of day variable
test(e2e): expect a summary list in Welsh overview tests
test(e2e): remove broken test for training case reference
test(e2e): fix smoke tests
test(e2e): allow whitespace padding when searching list keys by text
chore(web/applications): use summary-list for case preview info table
test(e2e): expect space after comma delimiter for regions
test(e2e): update tests for Welsh fields

## Issue ticket number and link

[APPLICS-509](https://pins-ds.atlassian.net/browse/APPLICS-509)

## Type of change 🧩

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update
- [ ] Other (please explain in the description section above)

## Checklist before requesting a review

- [x] I have performed a self-review of my own code
- [x] I have double checked this work does not include any hardcoded secrets or passwords
- [ ] I have made corresponding changes to the documentation
- [ ] I have provided details on how I have tested my code
- [x] I have referenced the ticket number above
- [x] I have provided a description of the ticket
- [x] I have included unit tests to cover any testable code changes

[APPLICS-509]: https://pins-ds.atlassian.net/browse/APPLICS-509?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by francescobrenci-AR — DISMISSED** [review]

File: apps/web/src/server/lib/nunjucks-filters/__tests__/build-table-row.test.js
Comment (BenSurgisonPINS) [inline apps/web/src/server/lib/nunjucks-filters/__tests__/build-table-row.test.js:1]: I think this file needs to be renamed to build-summary-list.test.js

File: apps/web/src/server/views/applications/components/case-form/case/title.njk [hunk 1]
```diff
@@ -1,36 +1,36 @@
...
 
 {% macro title(name, values, errors) %}
+	{% if name == 'titleWelsh' and values['title'] != null %}
 
-	{% set labelText = 'Name of the project' %}
-	{% if name == 'titleWelsh'%}
-		{% set labelText = 'Name of the project in Welsh' %}
-	{% endif %}
-
+    {% set labelText = 'Name of the project' %}
```
Comment (BenSurgisonPINS) [inline apps/web/src/server/views/applications/components/case-form/case/title.njk:7]: The label "Name of the project" should probably change to "Project name" to keep the english and welsh fields consistent.

File: apps/web/src/server/views/applications/components/case-form/case/title.njk
Comment (BenSurgisonPINS) [inline apps/web/src/server/views/applications/components/case-form/case/title.njk:1]: Not for this ticket, but I'm thinking a refactor at some point for all these text-area pages.  They could incorporate the macro textareaFormComponent as the styling, structure and decision to add English inset text is taken care of.  This macro was built while pairing with our designer.

**Review by BenSurgisonPINS — CHANGES_REQUESTED** [review]
Just a couple of minor changes and a suggestion (probably not for this PR)

**Review by BenSurgisonPINS — DISMISSED** [review]

**Review by francescobrenci-AR — APPROVED** [review]

**Review by BenSurgisonPINS — APPROVED** [review]
lgtm

**Pull request merged**